pelican/plugins/share_post/share_post.py

```diff
@@ -57,10 +59,16 @@ def create_link_facebook(title, url, content):
 
 @create_link
 def create_link_mastodon(title, url, content):
-    tags = getattr(content, "tags", [])
-    tags = ", ".join([tag.slug for tag in tags])
-    hashtags = f"&hashtags={tags}" if tags else ""
-    return f"https://toot.kytta.dev/?text={title}&url={url}{hashtags}"
+    if hasattr(content, "tags"):
+        taglist = content.tags
+        new_taglist = []
+        for i in taglist:
+            new_taglist.append("%23" + str(i))
+            hashtags = " ".join(str(x).replace(" ", "") for x in new_taglist)
+    else:
+        hashtags = ""
```

Comment:
```suggestion
    hashtags = ""
 
    if hasattr(content, "tags"):
        new_taglist = [f"%23{i.replace(" ", "")} for i in content.tags]
        hashtags = " ".join(new_taglist)
```

Can you see if this works? I'm not sure that having hashtags inside the for loop (your line 67) works correctly, but maybe I just misunderstood your code.